starfive/visionfive/v2/default.nix

```diff
@@ -1,21 +1,45 @@
-{ config, lib, pkgs, ... }: {
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+{
   boot = {
-    # Force no ZFS (from nixos/modules/profiles/base.nix) until updated to kernel 6.0
-    supportedFilesystems =
-      lib.mkForce [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "ntfs" "cifs" ];
     consoleLogLevel = lib.mkDefault 7;
-    kernelPackages = lib.mkDefault (pkgs.callPackage ./linux-6.6.nix {
-      inherit (config.boot) kernelPatches;
-    });
+    # Switch to default as soon they reach >= 6.11
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
 
-    kernelParams =
-      lib.mkDefault [ "console=tty0" "console=ttyS0,115200n8" "earlycon=sbi" ];
+    kernelParams = [
+      "console=tty0"
+      "console=ttyS0,115200n8"
+      "earlycon=sbi"
+    ];
 
     initrd.availableKernelModules = [ "dw_mmc_starfive" ];
 
+    # Support booting SD-image from NVME SSD
+    initrd.kernelModules = [
+      "dw_mmc-starfive"
+      "clk-starfive-jh7110-aon"
+      "clk-starfive-jh7110-stg"
+      "phy-jh7110-pcie"
+      "pcie-starfive"
+      "nvme"
+    ];
+
     loader = {
       grub.enable = lib.mkDefault false;
       generic-extlinux-compatible.enable = lib.mkDefault true;
     };
   };
+
+  hardware.deviceTree.name = lib.mkDefault "starfive/jh7110-starfive-visionfive-2-v1.3b.dtb";
```

Comment:
Last time I tested this shouldn't be set since u-boot would pass the correct dt name to kernel.